crawl.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime, timedelta
+from os import makedirs, path
+
+import pandas as pd
+import requests
+
+import reddit_auth
+
+TOKEN = reddit_auth.get_token()
+
+def fetch_posts(number:int = 3, page:str = None):
+  """fetches # of posts, starting at page reference"""
```

Comment:
[Nit] Missing return typehint and docstring could be clearer.

```suggestion
def fetch_posts(number: int = 3, page: str = None) -> Tuple[pd.DataFrame, Optional[str]]:
    """Fetches the number of posts starting at the page reference.

    Args:
      number: Max number of posts to fetch from API.
      page: The string ID of the page to fetch from reddit

    Returns: Dataframe of reddit posts from the given page and the next page ID if more posts exist on the next page.
    """
 ```
     
For the type hinting, `Optional` means the returned value can either be a `str` type or `None`.
    
`Tuple` and `Optional` in the code above need to be imported from the `typing` module ([python stdlib docs](https://docs.python.org/3.9/library/typing.html))
    
 ```python
    from typing import Tuple, Optional
```